tools/dgeni/common/decorators.ts

```diff
@@ -40,15 +41,21 @@ export function isNgModule(doc: ClassExportDoc) {
   return hasClassDecorator(doc, 'NgModule');
 }
 
-export function isDeprecatedDoc(doc: any) {
-  return ((doc.tags && doc.tags.tags) || []).some((tag: any) => tag.tagName === 'deprecated');
+export function isDeprecatedDoc(doc: ApiDoc) {
+  if (!isApiDocWithJsdocTags(doc)) {
+    return false;
+  }
+
+  return doc.tags.tags.some(tag => tag.tagName === 'deprecated');
 }
 
 /** Whether the given document is annotated with the "@docs-primary-export" jsdoc tag. */
-export function isPrimaryExportDoc(doc: any) {
-  return ((doc.tags && doc.tags.tags) || []).some(
-    (tag: any) => tag.tagName === 'docs-primary-export',
-  );
+export function isPrimaryExportDoc(doc: ApiDoc) {
```

Comment:
There are a few functions identical to this one. Maybe there should be a single function where you pass in the tag name you're looking for?

Reply:
yeah, good idea. adding that to the cleanup